internal/pkg/service/appsproxy/http/router.go

```diff
@@ -64,12 +68,18 @@ func NewRouter(d dependencies.ServiceScope, exceptionIDPrefix string) (*Router,
 		return nil, errors.PrefixError(err, "could not parse selection template")
 	}
 
+	transport, err := dns.NewDNSSkippingHTTPTransport(AppAddressCtxKey)
+	if err != nil {
```

Comment:
Suggestion, related to the previous comment:
- Move `transport.go`, to the `router` package (or create a new package).
- `transport = NewHTTPTransport(dns.NewDialer())` (+ error handling)